File: backend/Vms.Api/Domain/Entities/Recording.cs
using Vms.Api.Domain;

namespace Vms.Api.Domain.Entities;

public sealed class Recording
{
    public Guid Id { get; set; }

    public required string CameraId { get; set; }

    public Camera Camera { get; set; } = null!;

    public RecordingMode Mode { get; set; }

    public RecordingState State { get; set; }

    public required string FileName { get; set; }

    public DateTimeOffset StartedAt { get; set; }

    public DateTimeOffset? EndedAt { get; set; }

    public double? DurationSeconds { get; set; }

    public long? FileSizeBytes { get; set; }

    public string? FailureReason { get; set; }

    public Guid StartedByUserId { get; set; }

    public Guid? TriggerEventId { get; set; }

    public ICollection<RecordingKeyframe> Keyframes { get; set; } = [];
}

File: frontend/src/types/recording.ts
export type RecordingMode = 'Manual' | 'Continuous' | 'Event'
export type RecordingState = 'Recording' | 'Completed' | 'Failed'

export interface Recording {
  id: string
  cameraId: string
  cameraName: string
  mode: RecordingMode
  state: RecordingState
  startedAt: string
  endedAt: string | null
  durationSeconds: number | null
  fileSizeBytes: number | null
  failureReason: string | null
  triggerEventId: string | null
}

export interface RecordingCommand {
  message: string
  recording: Recording
}

export interface RecordingKeyframe {
  id: string
  timestampSeconds: number
}

export interface RecordingDetails {
  recording: Recording
  keyframes: RecordingKeyframe[]
}
